### DISCUSSION TITLE
Set base url and pre-populate broker and topics in web version?

### DISCUSSION THREAD

**Discussion by carlsa** [emqx/MQTTX#1446] [Q&A]
I'm wondering if is possible to

- Change the base URL for the web version so I can serve it at something like example.com/mqttx/ ?
- Serve a configuration file that pre-populate the web version with a default broker and MQTT topics?

My use case is that I want to serve _MQTTX web_ trough a reverse proxy in a docker compose file to allow someone to monitor the traffic from the other containers without having to set the broker and topics that the other containers are using.

**Comment by Red-Asuka** [comment]
For your first question, you can address it by modifying the project’s [`publicPath`](https://github.com/emqx/MQTTX/blob/main/web/vue.config.js#L8) and [`outputDir`](https://github.com/emqx/MQTTX/blob/main/web/vue.config.js#L9) configurations, and execute the build using the source code yourself. As for your second question, we currently do not provide this feature, but we will take into account community feedback and consider supporting it in future releases. Meanwhile, if you are interested, we welcome your contribution to this feature.

**Comment by drik98** [comment]
It's really a shame that we currently cannot preconfigure the default broker and mqtt topics. I would have loved to use this project in my IaC docker-compose setup.
